code/datums/helper_datums/teleport.dm

```diff
@@ -202,3 +191,27 @@
 	var/list/turfs = get_teleport_turfs(center, precision)
 	if (length(turfs))
 		return pick(turfs)
+
+/// Validates that the teleport being attempted is valid or not
+/proc/check_teleport_valid(atom/teleported_atom, atom/destination, channel)
+	var/area/teleported_atom_area = get_area(teleported_atom)
+	var/area/destination_area = get_area(destination)
+	var/turf/teleported_atom_turf = get_turf(teleported_atom)
+	var/turf/destination_turf = get_turf(destination)
```

Comment:
```suggestion
	var/area/origin_area = get_area(teleported_atom)
	var/turf/origin_turf = get_turf(teleported_atom)

	var/area/destination_area = get_area(destination)
	var/turf/destination_turf = get_turf(destination)
```

this grouping makes more sense (origin+origin, destination+destination). changing the name is optional but i think the "origin versus destination" dichotomy makes more sense than what is there currently

Reply:
sure